src/emulator/Emulator.js

```diff
@@ -213,6 +231,18 @@ class EmulatorImpl {
         }
     }
 
+    /*
+     * @param {TxOutputId} id
+     * @returns {Promise<TxInput>}
+     */
+    async hasUtxo(id) {
```

Comment:
I think it might be even cleaner to use the following one-liner instead: 

```
return id.toString() in this._allUtxos
```

Reply:
yep, it did turn out nicer that way.  Thanks